src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RADIUS_VALUE = 20;
+    private static final int MAX_QUANTITIES_FIGURES = 4;
+    private static final int MAX_RANDOM_VALUE = 20;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomRadius = random.nextInt(MAX_RADIUS_VALUE);
+        int randomBase1 = random.nextInt(MAX_RANDOM_VALUE);
+        int randomBase2 = random.nextInt(MAX_RANDOM_VALUE);
+        int height = random.nextInt(MAX_RANDOM_VALUE);
+        int firstLeg = random.nextInt(MAX_RANDOM_VALUE);
+        int secondLeg = random.nextInt(MAX_RANDOM_VALUE);
+        int leg = random.nextInt(MAX_RANDOM_VALUE);
+        int index = random.nextInt(MAX_QUANTITIES_FIGURES);
```

Comment:
Each time this method is called you generate random values for every case, but you need for one only. Let's do it inside case blocks